app/src/main/java/com/practium/playlistmaker/SearchActivity.kt

```diff
@@ -0,0 +1,87 @@
+package com.practium.playlistmaker
+
+import android.content.Context
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.widget.EditText
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+
+class SearchActivity : AppCompatActivity() {
+
+    companion object {
+        const val SEARCH_TEXT = "SEARCH_TEXT"
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+
+        var searchEditText = findViewById<EditText>(R.id.search_editeText)
+         outState.putString(SEARCH_TEXT, searchEditText.text.toString())
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+
+        var savedSearchText = savedInstanceState.getString(SEARCH_TEXT,"Пустота")
```

Comment:
🍏 Лучше в качестве дефолтного значения используй пустую строку
```suggestion
        var savedSearchText = savedInstanceState.getString(SEARCH_TEXT, "")
```